### PR TITLE
revert IPC server change + fix pipe communication on Windows

### PR DESCRIPTION
**Pull request by paul-marechal**
#### What it does

Revert https://github.com/eclipse-theia/theia/pull/11530

Use `overlapped` instead of `pipe` when opening a process pipe for the bidirectional communication to work properly on all platforms.

#### How to test

File watching should work on every platform: Try renaming files, etc.

#### Review checklist

- [x] As an author, I have thoroughly tested my changes and carefully followed [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#requesting-a-review)

#### Reminder for reviewers

- As a reviewer, I agree to behave in accordance with [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#reviewing)

### CODE REVIEW

**Comment by paul-marechal** [comment]
cc @tortmayr

File: .vscode/launch.json [hunk 1]
```diff
@@ -59,6 +59,7 @@
       "args": [
         "--hostname=0.0.0.0",
         "--port=3000",
+        "--no-cluster",
```
Comment (paul-marechal) [inline .vscode/launch.json:62]: Do we still want rid of that?

File: packages/core/src/node/messaging/ipc-channel.ts [hunk 2]
```diff
@@ -48,13 +49,12 @@ export class IPCChannel implements Channel {
 
     protected ipcErrorListener: (error: Error) => void = error => this.onErrorEmitter.fire(error);
 
-    constructor(pipe: Duplex, childProcess?: cp.ChildProcess) {
```
Comment (paul-marechal) [inline packages/core/src/node/messaging/ipc-channel.ts:51]: Do we still want to pass the duplex to listen to as argument?

Comment (tortmayr) [reply packages/core/src/node/messaging/ipc-channel.ts:51]: I don't think that is necessary.

File: .vscode/launch.json (hunk 1 above)
Comment (tortmayr) [reply .vscode/launch.json:62]: No, that was an oversight on my side. Thanks for readding that.

**Comment by tortmayr** [comment]
@paul-marechal Thanks for following up on that. Using  `overlapped` instead of `pipe` seems to fix all Windows issues. Definitely much cleaner than the named pipe setup introduced with  https://github.com/eclipse-theia/theia/pull/11530. :+1:

**Review by msujew — APPROVED** [review]
The changes look good to me 👍 I can confirm that the application behaves as expected on Windows.

File: .vscode/launch.json (hunk 1 above)
Comment (paul-marechal) [reply .vscode/launch.json:62]: The revert did it, not me :)

**Pull request merged**